contrib/natss/config/broker/natss.yaml

```diff
@@ -68,7 +68,7 @@ spec:
     spec:
       containers:
         - name: nats-streaming
-          image: nats-streaming:0.11.0
+          image: nats-streaming:0.12.2
```

Comment:
what I generally wonder is: do we really want to do vendoring of these 3rd party files ? 

For the apache kafka channel, we did point to some documentation that shows how to run it on k8s. Benefit: the tutorial/documentation is maintained by others, we just point to the link from our README.

Reply:
@evankanderson thoughts?

Reply:
We have here more aspects:
- NATSS could be used to have a simple, low-level, “battery-included” solution working perfectly on minikube, but usable also in a production environment.
- NATSS can be easy embedded in Go unit test (in process), see a very simple test:   https://github.com/knative/eventing/blob/master/contrib/natss/pkg/stanutil/stanutil_test.go
- The Go NATSS “client stub” is a little more complex as the “Sarama” stub.
- As an example, Kyma transparently installs NATSS as the default messaging broker for its embedded Knative/Eventing runtime environment. As a consequence, a Kyma developer can write functions without any NATSS knowledge.    
 
Perhaps a solution could be that “contrib/natss” should have its own “vendor” and to change Eventing to be "modular".